packages/client/components/AddTeamDialog.tsx

```diff
@@ -0,0 +1,201 @@
+    newUsers.forEach((user) => {
+      //add user.organizationIds to selectedUserOrganizationIds
+      user.organizationIds.forEach((organizationId) => {
+        selectedUsersOrganizationIds.add(organizationId)
+      })
+    })
+
+    const mutualOrgs = viewerOrganizations.filter((org) => selectedUsersOrganizationIds.has(org.id))
+
+    const mutualOrgsIds = mutualOrgs.map((org) => org.id)
+    setMutualOrgsIds(mutualOrgsIds)
+    setSelectedOrgId(mutualOrgsIds[0] ?? viewerOrganizations[0]?.id)
+
+    if (!teamNameManuallyEdited) {
+      setTeamName(generateTeamName(newUsers))
+    }
+  }
+
+  const handleAddTeam = () => {
+    const newTeam = {
+      name: teamName.substring(0, MAX_TEAM_NAME_LENGTH),
+      orgId: selectedOrgId
+    }
+    submitMutation()
+
+    AddTeamMutation(
+      atmosphere,
+      {newTeam, invitees: selectedUsers.map((user) => user.email)},
+      {
+        onError,
+        onCompleted: (res, errors) => {
+          onCompleted(res, errors)
+          const error = getGraphQLError(res, errors)
+          if (!error) {
+            onAddTeam(res.addTeam.team.id)
+          }
+        },
+        history,
+        showTeamCreatedToast: false
+      }
+    )
+  }
+
+  const isValid = selectedUsers.length && teamName.trim()
+
+  return (
+    <Dialog isOpen={isOpen} onClose={onClose}>
+      <DialogContent className='z-10'>
+        <DialogTitle className='mb-4'>Add team</DialogTitle>
+
+        <fieldset className={fieldsetStyles}>
+          <label className={labelStyles}>Add teammates</label>
+
+          <AdhocTeamMultiSelect
+            viewerRef={viewer}
+            value={selectedUsers}
+            onChange={onSelectedUsersChange}
+          />
+
+          {selectedUsers.some((user: Option) => !user.id) && (
+            <div className='mt-3 text-xs font-semibold text-slate-700'>
+              Email invitations expire in 30 days.
+            </div>
+          )}
+        </fieldset>
+
+        {showOrgPicker && (
+          <fieldset className={fieldsetStyles}>
+            <label className={labelStyles}>Organization</label>
+            <Select onValueChange={(orgId) => setSelectedOrgId(orgId)} value={selectedOrgId}>
+              <SelectTrigger className='bg-white'>
+                <SelectValue />
+              </SelectTrigger>
+              <SelectContent>
+                <SelectGroup>
+                  {viewerOrganizations
+                    .filter((org) => (mutualOrgsIds.length ? mutualOrgsIds.includes(org.id) : true))
+                    .map((org) => (
+                      <SelectItem value={org.id} key={org.id}>
+                        {org.name}
+                      </SelectItem>
+                    ))}
+                </SelectGroup>
+              </SelectContent>
+            </Select>
+          </fieldset>
+        )}
+
+        <fieldset className={fieldsetStyles}>
+          <label className={labelStyles}>Team name</label>
+          <Input
```

Comment:
-1 we're enforcing a team name length limit of 50 characters, but not giving users feedback that that will happen. So if I create a team with 55 characters, the last 5 characters get removed without me knowing.

We could add `maxLength={MAX_TEAM_NAME_LENGTH}` to the input. Or, for the optimal UX, we could give the user some feedback that there's a limit.

Reply:
I just removed this now, and allow backend to validate this. I think rarely someone would type 50 chars